src/main/java/io/jenkins/plugins/audit/listeners/ItemChangeListener.java

```diff
@@ -0,0 +1,137 @@
+package io.jenkins.plugins.audit.listeners;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.model.Item;
+import hudson.model.User;
+import io.jenkins.plugins.audit.event.*;
+import org.apache.logging.log4j.audit.LogEventFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Date;
+
+/**
+ * Listener notified of any CRUD operations on items.
+ */
+@Extension
+public class ItemChangeListener extends hudson.model.listeners.ItemListener{
+
+    /**
+     * Fired when a new job is created and added to Jenkins, before the initial configuration page is provided.
+     *
+     * @param item is a job that is created
+     */
+    @Override
+    public void onCreated(@Nonnull Item item) {
+        CreateItem itemCreateEvent = LogEventFactory.getEvent(CreateItem.class);
+
+        itemCreateEvent.setItemName(item.getName());
+        itemCreateEvent.setItemUri(item.getUrl());
+        itemCreateEvent.setTimestamp(new Date().toString());
+        User user = User.current();
+        if(user != null)
+            itemCreateEvent.setUserId(user.getId());
+        else
+            itemCreateEvent.setUserId(null);
```

Comment:
No need to set `null` here. I'm not really sure when User will be null anyways, though in that scenario, it might make more sense to use the system user as the default.

As a matter of fact, this might be more appropriate to add to the servlet filter class in https://github.com/jenkinsci/audit-log-plugin/pull/18 so you don't need to keep repeating yourself.

Reply:
Does it mean that user need not be set and be set by #18 all the time to all the events? I am little confused about this. I copy pasted it from another event (BuildStartListener) where they set the User. Can you please explain?

Reply:
The #18 PR purpose is to create a servlet filter which is executed for every HTTP request. This filter can be used to automatically set some common `RequestContext` data which is common to the full request. Then you don't need to manually add an attribute to the audit event since every event will contain the common/shared attributes. Since all audit events happen on behalf of a principal (user), we want to associate that userid to the event.

Reply:
I made the userId optional for now and trying to get it set from the #18.